src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs

```diff
@@ -327,10 +330,24 @@
                           :reload-after-delete #(this :-get-clusters-list-if-whitelisted))]))]))
    :component-did-mount
    (fn [{:keys [this]}]
-     (this :-is-leo-whitelisted))
+     (do (this :-is-leo-whitelisted)
+         (.addEventListener js/window "message" (this :-handle-postmessage-event))))
+
    :component-will-unmount
-   (fn [{:keys [locals]}]
-     (swap! locals assoc :dead? true))
+   (fn [{:keys [this locals]}]
+     (do (swap! locals assoc :dead? true)
+         (.removeEventListener js/window "message" (this :-handle-postmessage-event))))
+
+   ; Communicates with the Leo notebook extension
+   :-handle-postmessage-event
+   #(fn [e]
```

Comment:
Is this correct -- i.e. will (this :-handle-postmessage-event) return the same function reference for both addEventListener and removeEventListener?

Reply:
Probably? Our pattern has been to store the function in `locals`.

Reply:
Seems like this was _not_ working -- the listener stuck around after unmount was called. Changing it to store in `locals` and it's working as expected.

Reply:
`(this :foo)` calls the method foo on the component. `(react/method this :foo)` returns a stable binding to the foo method.

Reply:
I will try `react/method`!